frontend/src/metabase/visualizations/visualizations/SmartScalar/SmartScalar.jsx

```diff
@@ -96,11 +96,10 @@ export function SmartScalar({
     }
   };
 
-  const { displayValue, fullScalarValue } = compactifyValue(
-    value,
-    width,
-    formatOptions,
-  );
+  const { displayValue, fullScalarValue } = compactifyValue(value, width, {
+    ...formatOptions,
+    compact: settings["scalar.compact_primary_number"],
+  });
```

Comment:
Is it fine to set `compact` here? Should that maybe happen somewhere around [here](https://github.com/metabase/metabase/blob/fe2b937807065f056d917f38021f8ff3082b7323/frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.js#L23) in `computeTrend`? 🤔

Reply:
@JesseSDevaney thanks! Your suggestion makes a ton of sense; I think I went the same path, but without changing `compactifyValue.` I think it should be enough as long as `compactifyValue` respects the `formatOptions.compact` setting. I would avoid optimizing it for our use case because it is [also shared with a "number" visualization](https://github.com/metabase/metabase/blob/913c0b04554a9e8b9e3c112e10c6c3c3274e8f33/frontend/src/metabase/visualizations/visualizations/Scalar/Scalar.jsx#L189)